src/calendar/tests/CalendarDate.spec.js

```diff
@@ -0,0 +1,460 @@
+import React from 'react/addons';
+import moment from 'moment';
+
+import CalendarDate from '../CalendarDate.jsx';
+
+import CalendarDatePeriod from '../CalendarDatePeriod.jsx';
+import CalendarHighlight from '../CalendarHighlight.jsx';
+import CalendarSelection from '../CalendarSelection.jsx';
+
+const TestUtils = React.addons.TestUtils;
+
+describe('The CalendarDate Component', function () {
+
+  const getCalendarDate = (props) => {
+
+    props = props || {};
+
+    return (<CalendarDate
+      date={props.date || moment()}
+      firstOfMonth={props.firstOfMonth || moment()}
+      dateRangesForDate={function () {
+        return {
+          count: function () {
+            return props.count || 1;
+          },
+          getIn: function (data) {
+            if (data[0] === 0) {
+              return '#333';
+            }
+            return '#444';
+          },
+        };
+      }}
+      onSelectDate={this.selectDateSpy}
+      onHighlightDate={this.highlightDateSpy}
+      onUnHighlightDate={this.unHighlightDateSpy}
+      isHighlightedDate={props.isHighlightedDate || false }
```

Comment:
Should our underlying component not have a reliable implementation of `getDefaultProps` - or are you worried about a shifting implementation breaking these tests inadvertantly?

If so, maybe we could add underscore as a test dependency and use `_.defaults` or `_.extend` to provide these default values, then slice in all of the required defaulted props

```javascript
var defaultedProps = _.extend({
  isHighlightedDate: false,
  etc, etc
}, props)


return (
    <CalendarDate {...defaultedProps} />
)
```